packages/@tailwindcss-upgrade/src/template/codemods/important.ts

```diff
@@ -19,9 +19,44 @@ export function important(
   designSystem: DesignSystem,
   _userConfig: Config,
   rawCandidate: string,
+  location?: {
+    contents: string
+    start: number
+    end: number
+  },
 ): string {
   for (let candidate of parseCandidate(rawCandidate, designSystem)) {
     if (candidate.important && candidate.raw[candidate.raw.length - 1] !== '!') {
+      // The important migration is one of the most broad migrations with a high
+      // potential of matching false positives since `!` is a valid character in
+      // most programming languages. Since v4 is technically backward compatible
+      // to v3 in that it can read `!` in the front of the utility too, we err
+      // on the side of caution and only migrate candidates that we are certain
+      // are inside of a string.
+      if (location) {
+        let charBefore = location.contents.at(location.start - 1)
+        let charAfter = location.contents.at(location.end)
+
+        if (
+          charBefore !== undefined &&
+          charBefore !== '"' &&
+          charBefore !== "'" &&
+          charBefore !== '`' &&
+          charBefore !== ' '
+        ) {
+          continue
+        }
+        if (
+          charAfter !== undefined &&
+          charAfter !== '"' &&
+          charAfter !== "'" &&
+          charAfter !== '`' &&
+          charAfter !== ' '
+        ) {
+          continue
+        }
+      }
```

Comment:
Could we in theory scan backwards and forwards to look for quotes _on the same line? I feel like this logic may be too restrictive.

Reply:
liked that, implemented 😎

Reply:
I think that change disappeared — don't see it in the diff 😬